db/app/app.test.js

```diff
@@ -91,14 +90,138 @@ describe("get /api/articles", () => {
+        expect(article).toHaveProperty("author");
+        expect(article).toHaveProperty("article_id");
+        expect(article).toHaveProperty("body");
+        expect(article).toHaveProperty("created_at");
+        expect(article).toHaveProperty("votes");
+        expect(article).toHaveProperty("article_img_url");
+      });
+  });
+
+  test("checks the type of the values", () => {
+    return request(app)
+      .get("/api/articles/3")
+      .then((res) => {
+        let article = res.body.article;
+        expect(typeof article.title).toBe("string");
+        expect(typeof article.topic).toBe("string");
+        expect(typeof article.author).toBe("string");
+        expect(typeof article.article_id).toBe("number");
+        expect(typeof article.body).toBe("string");
+        expect(typeof article.created_at).toBe("string");
+        expect(typeof article.votes).toBe("number");
+        expect(typeof article.article_img_url).toBe("string");
+      });
+  });
+  test("checks whether the articles are ordered by date created", () => {
+    return request(app)
+      .get("/api/articles")
+      .then((res) => {
+        expect(res.body.articles).toBeSortedBy("created_at", {
+          descending: true,
+        });
+      });
+  });
+
+  test("400: Should respond with Bad Request for invalid id", () => {
+    const id = "dog";
+
+    return request(app)
+      .get(`/api/articles/${id}`)
+      .expect(400)
+      .then((res) => {
+        expect(res.body.msg).toBe("Bad Request");
+      });
+  });
+
+  test("404: Should respond with Not Found for id not in database ", () => {
+    return request(app)
+      .get(`/api/articles/999`)
+      .expect(404)
+      .then((res) => {
+        expect(res.body.msg).toBe("Not Found");
+      });
+  });
+});
 
-   test("checks whether the articles are ordered by date created", () => {
-     return request(app)
-       .get("/api/articles")
-       .then((res) => {
-          expect(res.body.articles).toBeSortedBy("created_at", {
-            descending: true
-          });
-       });
-   });
+
+describe("get /api/articles/:id/comments", () => {
+  test("responds with status 200", () => {
+    return request(app).get("/api/articles/3/comments").expect(200);
+  });
+
+  test("checks whether each article has the correct keys", () => {
+    return request(app)
+      .get("/api/articles/3/comments")
+      .then((res) => {
+        res.body.comments.forEach((comment) => {
+          expect(comment).toHaveProperty("comment_id");
+          expect(comment).toHaveProperty("votes");
+          expect(comment).toHaveProperty("created_at");
+          expect(comment).toHaveProperty("author");
+          expect(comment).toHaveProperty("body");
+          expect(comment).toHaveProperty("article_id");
+        });
+      });
+  });
+
+  test("checks the type of the values", () => {
+    return request(app)
+      .get("/api/articles/3/comments")
+      .then((res) => {
```

Comment:
Small thing. Be aware when using forEach in your assertions. If for some reason whatever came back from our database was an empty array, forEach would not execute and we would never run our assertions. 

This would leave to a false positive. We can negate this by checked that the length of what is returned is what we expect it to be - that way if something empty comes back - we would fail that assertion and not have to worry about the forEach